lib/meta.nix

```diff
@@ -87,5 +87,5 @@ rec {
         else { parsed = elem; };
     in lib.matchAttrs pattern platform;
 
-  enableIfAvailable = p: if p.meta.available or true then [ p ] else [];
+  enableIfSupported = p: lib.optional p.meta.supported or true p;
```

Comment:
I'm not sure if parens are literally required here but `lib.optional (p.meta.supported or true) p` seems much better IMO.